CMakeLists.txt

```diff
@@ -187,7 +187,7 @@ target_compile_features(geos_cxx_flags INTERFACE cxx_std_11)
 target_compile_options(geos_cxx_flags INTERFACE
 	"$<$<OR:$<CXX_COMPILER_ID:Clang>,$<CXX_COMPILER_ID:AppleClang>>:-ffp-contract=off>"
 	"$<$<CXX_COMPILER_ID:GNU>:-ffp-contract=off>"
-	"$<$<CXX_COMPILER_ID:MSVC>:/fp:precise>"
+	"$<$<BOOL:${MSVC}>:/fp:precise>"
```

Comment:
Why the bool thing? Is the old way broken? (I find the old one more straightforward in my head.)

Reply:
Nevermind, you're picking off mingw and msvc with one shot, got it. Add a comment to that effect?

Reply:
Unrelated to mingw. AFAIU `$<CXX_COMPILER_ID:MSVC>` doesn't cover clang-cl, but `${MSVC}` does.